Herebyfile.mjs

```diff
@@ -614,6 +616,105 @@ export const runTests = task({
 //     "   --shardId": "1-based ID of this shard (default: 1)",
 // };
 
+export const runTestsAndWatch = task({
+    name: "runtests-watch",
+    dependencies: [watchTests],
+    run: async () => {
+        if (!cmdLineOptions.tests && !cmdLineOptions.failed) {
+            console.log(chalk.redBright(`You must specifiy either --tests/-t or --failed to use 'runtests-watch'.`));
+        }
+
+        let watching = true;
+        let running = true;
+        let testsChangedDeferred = /** @type {Deferred<void>} */(new Deferred());
+        let testsChangedAbortController = new AbortController();
+        const testsChangedDebouncer = new Debouncer(1_000, async () => {
+            testsChangedDeferred.resolve();
+            testsChangedDeferred = /** @type {Deferred<void>} */(new Deferred());
+        });
+
+        // catch SIGINT to ensure we can kill the watcher
+        const rl = createInterface(process.stdin);
+        rl.on("SIGINT", () => {
+            if (running) {
+                console.log(chalk.yellowBright("[watch] aborting in-progress test run..."));
+            }
+
+            testsChangedAbortController.abort();
+            testRunnerWatcher.close();
+            testCaseWatcher.close();
+            watching = false;
+        });
+        process.on("beforeExit", () => {
+            console.log(chalk.yellowBright("[watch] exiting watch mode..."));
+        });
+
+        const testRunnerWatcher = chokidar.watch([
+            testRunner

```

Comment:
This file currently gets updated twice to replace esbuild's `__require` function with the real one. Does that cause any problems with chokidar or is all of the debounce logic handling that?

Reply:
Debounce should handle it, depending on how long that takes. Either way it will kill the test task if it starts too soon. If it comes down to it, I can use the change you proposed to handle run.js

Reply:
Should be basically instant, so that's probably fine.